### PR TITLE
Upgrade to go 1.14 and go.mod

### PR DESCRIPTION
**Pull request by antechrestos**
Stuck at home due to covid restriction, I finally got time to wrote this contribution :smirk:

### CODE REVIEW

**Comment by antechrestos** [comment]
@nkubala My bad. I've just pushed a new version.
I based my work on what I saw on other projects layout and on this [article](https://eli.thegreenplace.net/2019/simple-go-project-layout-with-modules/)

**Comment by nkubala** [comment]
@antechrestos ah makes sense I see what you were doing. yeah this is more in line with what we do in https://github.com/GoogleContainerTools/skaffold - code related to the actual commands goes in `cmd/skaffold`, and the actual business logic (like 90% of the code) lives in `pkg/skaffold`. this project was created way back before I knew about best practices around go project structure 😄 

I'm wondering...would you be willing to try and split some code out into `cmd/container-diff` and `pkg/container-diff` to clean things up a bit? that would go a long way towards future contributions - I think the code as is right now is a bit hard to follow and would be more approachable that way. totally understand if this is more than you want to take on right now, but I figured I'd ask 😃  my team isn't going to be able to prioritize it any time soon.

either way, I'd love if you could split out any refactors into a separate PR so we can get the restructure/naming right there. then this PR will be just go 1.14 related stuff and I'm happy to merge it!

thanks for giving this project a little bit of love 🙏

**Comment by antechrestos** [comment]
@nkubala would you agree doing the contrary: forget any renaming / move of this MR and just go to go 1.14 and **go mod** then I can clean things? 

It would allow us to benefit go module. Then I can start cleaning things.

If you agree, this merge requests only contains changes relative to upgrading to go 1.14

**Comment by nkubala** [comment]
@antechrestos yep that sounds good to me!

**Comment by antechrestos** [comment]
@nkubala I've rework my branch to split the migration to golang 1.4 in one commit and the go module one to another commit. This branch does not brings any other change but these one.
